### ISSUE TITLE
Allow for additional PageData properties

### ISSUE THREAD

**Issue by paulboocock** [DavidWells/analytics#66]
The current implementation allows for the PageData attributes to be overridden when calling `analytics.page({...})`.

It would be beneficial if additional properties could be specified in PageData beyond the standard. 

And example of this, is adding additional Context to a Page View event with Snowplow.

For Snowplow, a common use case would be to inject the array of context data. This would look something like:

```
analytics.page({
  context: [{
    schema: 'iglu:com.acme/blog_post/jsonschema/1-0-0',
    data: {
      title: 'Re-thinking the structure of event data',
      category: 'Data Insights',
      datePublished: '2020-01-24'
    }
  }]
});
```

**Comment by DavidWells** [comment]
Hey @paulboocock 

This is possible with the `page` method today.

```js
analytics.page({
  context: [{
    schema: 'iglu:com.acme/blog_post/jsonschema/1-0-0',
    data: {
      title: 'Re-thinking the structure of event data',
      category: 'Data Insights',
      datePublished: '2020-01-24'
    }
  }]
});

/* will result in 
{
  "properties": {
    "title": "Analytics Demo",
    "url": "http://localhost:3000/",
    "path": "/",
    "hash": "",
    "search": "",
    "width": 1680,
    "height": 515,
    "referrer": "http://localhost:3000/",
    "context": [{
      "schema": "iglu:com.acme/blog_post/jsonschema/1-0-0",
      "data": {
        "title": "Re-thinking the structure of event data",
        "category": "Data Insights",
        "datePublished": "2020-01-24"
      }
    }]
  },
  "options": {},
  "userId": null,
  "anonymousId": "ccb1bd6c-0349-4b9b-a8c2-01e3e4f3398a",
  "meta": {
    "timestamp": 1595916737999
  }
}
*/
```

Then inside of a plugin or listener, you can use `properties.context` values.

Here is a plugin example accessing the custom context values

```js
import Analytics from 'analytics'

const myCustomPlugin = {
  name: 'plugin-1',
  page: ({ payload }) => {
    console.log('page view payload', payload.properties)
    console.log('page view context', payload.properties.context)
  }
}

/* initialize analytics and load plugins */
const analytics = Analytics({
  app: 'cool-apps',
  plugins: [
    myCustomPlugin,
  ]
})

export default analytics
```

**Comment by paulboocock** [comment]
D'oh!

I put my call to analytics.page() in the button click handler only in the Demo Example so I wasn't getting it on the initial page  event, only the one when clicking the button.

I was rather confused when I opened this ticket, as I'd already written the code for passing properties into the page event but couldn't get it to work when I went back to test it. 

Thanks for the effort in the reply. Knowing it should work stopped me going insane :-)
